src/js/node/https.ts

```diff
@@ -1,14 +1,30 @@
 // Hardcoded module "node:https"
 const http = require("node:http");
+const { urlToHttpOptions } = require("node:url");
 
-function request(input, options, cb) {
-  if (input && typeof input === "object" && !(input instanceof URL)) {
-    input.protocol ??= "https:";
-  } else if (typeof options === "object") {
-    options.protocol ??= "https:";
+const ObjectSetPrototypeOf = Object.setPrototypeOf;
+const ArrayPrototypeShift = Array.prototype.shift;
+const ObjectAssign = Object.assign;
+const ArrayPrototypeUnshift = Array.prototype.unshift;
+
+function request(...args) {
```

Comment:
Why is this code duplicated in http and https?

Reply:
this one assigns `options._defaultAgent`